subprojects/internal-testing/src/main/groovy/org/gradle/testing/internal/util/RetryRule.groovy

```diff
@@ -102,6 +102,7 @@ class RetryRule implements MethodRule {
     private void reRunSetup(SpecInfo spec) {
         if (spec != null) {
             reRunSetup(spec.getSuperSpec())
+            spec.getInitializerMethod()?.invoke(specification)
```

Comment:
❌ This probably failed because it doesn't match what Spock's runner does - First run all initializers in the hierarchy, then run all the setup methods.

See https://github.com/spockframework/spock/blob/d20d8e7d310f3678d92b10977af0ae4a8a460a5c/spock-core/src/main/java/org/spockframework/runtime/BaseSpecRunner.java

Reply:
I am not sure that is the only problem. Note that by re-running the initializers we also replace all the rule instances of the class.
I don't think spock will re-discover the rule instances from our resetted `specification`.
That means that the `specification` has different rule instances to the ones Spock is currently executing. So referencing those instances will lead to problems.

I can try to turn around the calls here and re-run the performance tests, though.

Reply:
Like I expected, that didn't help: https://builds.gradle.org/viewLog.html?buildId=11181999&tab=queuedBuildOverviewTab

Reply:
Damn, so we need to dig even deeper into how spock handles rules or handle them ourselves I guess?